subsys/bluetooth/mesh/transport.c

```diff
@@ -798,8 +800,8 @@ static struct seg_tx *seg_tx_lookup(uint16_t seq_zero, uint8_t obo, uint16_t add
 		 * acknowledgement, assume it's a Friend that's
 		 * responding and therefore accept the message.
 		 */
-		if (obo && tx->nack_count == tx->seg_n + 1) {
-			tx->dst = addr;
```

Comment:
Is it OK to just annotate this sentence? Without adding additional `ack_src`?

Reply:
This is to cover the last sentence in this paragraph (section 3.5.3.3):
> If a Segment Acknowledgment message OBO field is set to 1, and the DST field is set to the unicast address of this element, and the SeqZero field is set to SeqZero of the segmented message, the Segment Acknowledgment message is a valid acknowledgment for the segmented message. **For a given SeqAuth, only the Segment Acknowledgment messages from the first SRC address received should be considered valid**.